docs/guidelines.md

```diff
@@ -206,7 +207,9 @@ The punctuation placement rules can be summarized as:
   desirable.
 - Use the constructor initializer list and make sure its order matches the
   declaration order.
-- Prefer pre-increment/decrement (`++x`) to post-increment/decrement (`x++`)
+- Prefer pre-increment/decrement (`++x`) to post-increment/decrement (`x++`).
+  For objects this avoids creating a unused copy of the original object. It's
+  also recommended to do this for primitives just to enforce the habit.
```

Comment:
We didn't necessarily write justifications for all of these (that might turn in to a much larger document).  In this case we could de-emphasize the performance angle, since compilers can figure that out anyway.  It's a logically simpler operation -- the similar rule is in C++ core guidelines and https://google.github.io/styleguide/cppguide.html

Reply:
I think it would be nice, but I'm fine with leaving out a justification. My issue was I read this originally I thought it was just talking about iterators. How about:

```
- Prefer pre-increment/decrement (`++x`) to post-increment/decrement (`x++`) for both objects and non-objects.
```

Also I could find a rule in the [Google Guidelines](https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement), but not in the Core guidelines.

Reply:
I searched for "prefix" on http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

Reply:
I missed [that](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p9-dont-waste-time-or-space). I was looking for it as its own rule.